url21b471891f_Agetac_Model_tgz-pJ8-org_json_TestJ8/src/org/agetac/common/model/impl/Implique.java

```diff
@@ -0,0 +1,63 @@
+package org.agetac.common.model.impl;
+
+import org.agetac.common.exception.InvalidJSONException;
+import org.agetac.common.model.sign.AbstractModel;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class Implique extends AbstractModel {
+
+	public enum EtatImplique {
+	    INDEMNE, URGENCE_RELATIVE, URGENCE_ABSOLUE, DECEDE 
+	}
+	
+	private EtatImplique etat;
+
+	public Implique(String uid, EtatImplique etat) {
+		super(uid, null, null);
+		this.etat = etat;
+	}
+
+	public Implique(JSONObject json) throws InvalidJSONException {
+		super(json);
+		
+		try {
+			this.etat = EtatImplique.valueOf(json.getString("etat"));
+		} catch (JSONException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)